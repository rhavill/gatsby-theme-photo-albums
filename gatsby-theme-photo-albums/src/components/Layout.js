import React from 'react'
import PropTypes from 'prop-types'
import Header from './Header'
import {Layout as ThemeLayout, Main} from 'theme-ui'

const Layout =  ({children, path}) => (
  <ThemeLayout>
    <Header path={path}/>
    <Main>
      {children}
    </Main>
  </ThemeLayout>
)

Layout.propTypes = {
  path: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
}

export default Layout