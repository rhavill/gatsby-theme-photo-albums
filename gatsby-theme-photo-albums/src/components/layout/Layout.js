import React from 'react'
import PropTypes from 'prop-types'
import Header from './Header'
import LayoutStyles from '../../styles/Layout'
import MainContentStyles from '../../styles/MainContent'

const Layout = ({children, path}) => 
  <LayoutStyles>
    <Header path={path}/>
    <MainContentStyles>
      {children}
    </MainContentStyles>
  </LayoutStyles>

Layout.propTypes = {
  path: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
}

export default Layout