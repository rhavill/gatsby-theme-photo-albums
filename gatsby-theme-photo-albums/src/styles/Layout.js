/** @jsx jsx */
import { jsx } from 'theme-ui'
import PropTypes from 'prop-types'

const Layout = ({children}) => 
  <div sx={{variant: 'layout.root'}}>
    {children}
  </div>

Layout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
}

export default Layout