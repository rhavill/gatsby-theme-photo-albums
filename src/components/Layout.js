import React from 'react'
import PropTypes from 'prop-types'
import Header from './Header'
import {Layout as ThemeLayout} from 'theme-ui'

const Layout =  ({children, path}) => (
  <ThemeLayout>
    <Header path={path}/>
    <div className='container page'>
      {children}
    </div>
  </ThemeLayout>
)

Layout.propTypes = {
  path: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
}

export default Layout