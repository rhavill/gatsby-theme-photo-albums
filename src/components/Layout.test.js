import React from "react"
import renderer from "react-test-renderer"
import {cleanup, render} from '@testing-library/react';
import Layout from "./Layout"

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

describe("Layout", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Layout path="/"><div>hello</div></Layout>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("displays child nodes", () => {
    const {getByText} = render(
      <Layout path="/"><div>hello</div></Layout>
    );
    expect(getByText('hello')).toBeTruthy()
  })
})
