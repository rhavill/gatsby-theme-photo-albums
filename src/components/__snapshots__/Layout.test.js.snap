// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="wrapper"
>
  <header>
    <nav
      className="container"
    >
      <a
        href="/"
      >
        Home
      </a>
    </nav>
  </header>
  <div
    className="container"
  >
    <div>
      hello
    </div>
  </div>
</div>
`;
